include/uri/up_uri.h

```diff
@@ -73,13 +83,30 @@ class up_uri {
            ", uResource=" + uResource.tostring() + '}';
   }
 
+  size_t getHash() const
+  {
+      return _hash;
+  }
+  
+  std::string getBase64() const
+  {
+      return _base64uri;
+  }
+
+  std::string getTopic() const
+  {
+      return "";
+  }
+
  private:
   const uri_authority uAuthority;
   const uri_entity uEntity;
   const uri_resource uResource;
+  size_t _hash;
+  std::string _base64uri;
```

Comment:
use _base64uri_

Reply:
i am going to revert this as it should not be part of the PR